### PR TITLE
Update vendors for s390x support. Remove Solaris support.

### PR DESCRIPTION
**Pull request by creack**
Update goselect version to include support for s390x & co.
Update golang.org/x/crypto for the same reason.

Commands used:

```sh
$> export GO111MODULE=on
$> go get github.com/creack/goselect@v0.1.0
$> go get -u golang.org/x/crypto
$> go mod tidy && go mod vendor
```

Tested on IBM's free mainframe VM: https://developer.ibm.com/linuxone/

```sh
$> uname -a
Linux gotest 3.10.0-957.5.1.el7.s390x #1 SMP Wed Dec 19 10:46:37 EST 2018 s390x s390x s390x GNU/Linux
$> go version
go version go1.12.4 linux/s390x
$> PACKER_DEV=true make bin

WARN: 'make bin' is for debug / test builds only. Use 'make release' for release builds.
==> Checking for necessary tools...
==> Entering Packer source dir...
==> Ensuring output directories are present...
==> Removing old builds...
==> Building...
Number of parallel builds: 1

-->     linux/s390x: github.com/hashicorp/packer
==> Copying binaries for this platform...
‘./pkg/linux_s390x/packer’ -> ‘bin/packer’
‘./pkg/linux_s390x/packer’ -> ‘/home/linux1/go/bin/packer’

==> Results:
total 118M
-rwxrwxr-x 1 linux1 linux1 118M Apr 25 15:58 packer
```

Added mips, mips64, mipsle, mips64le and s390x in the cross compile build script.

Removed solaris/amd64 as it fails to compile on master (github.com/gofrs/flock and github.com/mattn/go-tty).

```sh
$> go version
go version go1.12.4 linux/amd64
```

Running `make bin` on master results in:

```sh
--> solaris/amd64 error: exit status 2
Stderr: # github.com/hashicorp/packer/vendor/github.com/gofrs/flock
vendor/github.com/gofrs/flock/flock_unix.go:28:22: undefined: syscall.LOCK_EX
vendor/github.com/gofrs/flock/flock_unix.go:39:22: undefined: syscall.LOCK_SH
vendor/github.com/gofrs/flock/flock_unix.go:56:12: undefined: syscall.Flock
vendor/github.com/gofrs/flock/flock_unix.go:66:12: undefined: syscall.Flock
vendor/github.com/gofrs/flock/flock_unix.go:96:12: undefined: syscall.Flock
vendor/github.com/gofrs/flock/flock_unix.go:96:42: undefined: syscall.LOCK_UN
vendor/github.com/gofrs/flock/flock_unix.go:118:21: undefined: syscall.LOCK_EX
vendor/github.com/gofrs/flock/flock_unix.go:130:21: undefined: syscall.LOCK_SH
vendor/github.com/gofrs/flock/flock_unix.go:149:9: undefined: syscall.Flock
vendor/github.com/gofrs/flock/flock_unix.go:149:44: undefined: syscall.LOCK_NB
vendor/github.com/gofrs/flock/flock_unix.go:149:44: too many errors
# github.com/hashicorp/packer/vendor/github.com/mattn/go-tty
vendor/github.com/mattn/go-tty/tty_unix.go:40:35: undefined: syscall.SYS_IOCTL
vendor/github.com/mattn/go-tty/tty_unix.go:40:76: undefined: ioctlReadTermios
vendor/github.com/mattn/go-tty/tty_unix.go:46:35: undefined: syscall.SYS_IOCTL
vendor/github.com/mattn/go-tty/tty_unix.go:46:76: undefined: ioctlWriteTermios
vendor/github.com/mattn/go-tty/tty_unix.go:67:32: undefined: syscall.SYS_IOCTL
vendor/github.com/mattn/go-tty/tty_unix.go:67:73: undefined: ioctlWriteTermios
vendor/github.com/mattn/go-tty/tty_unix.go:78:35: undefined: syscall.SYS_IOCTL
vendor/github.com/mattn/go-tty/tty_unix.go:93:57: undefined: ioctlReadTermios
vendor/github.com/mattn/go-tty/tty_unix.go:106:51: undefined: ioctlWriteTermios
vendor/github.com/mattn/go-tty/tty_unix.go:111:52: undefined: ioctlWriteTermios
vendor/github.com/mattn/go-tty/tty_unix.go:111:52: too many errors
```

Result of `make bin` with the added archs and solaris removed:

```sh
$> make bin
WARN: 'make bin' is for debug / test builds only. Use 'make release' for release builds.
==> Checking for necessary tools...
==> Entering Packer source dir...
==> Ensuring output directories are present...
==> Removing old builds...
==> Building...
Number of parallel builds: 31

-->   openbsd/amd64: github.com/hashicorp/packer
-->     windows/386: github.com/hashicorp/packer
-->      linux/mips: github.com/hashicorp/packer
-->       linux/arm: github.com/hashicorp/packer
-->       linux/386: github.com/hashicorp/packer
-->    linux/mipsle: github.com/hashicorp/packer
-->     linux/arm64: github.com/hashicorp/packer
-->   linux/ppc64le: github.com/hashicorp/packer
-->     linux/amd64: github.com/hashicorp/packer
-->   freebsd/amd64: github.com/hashicorp/packer
-->     freebsd/arm: github.com/hashicorp/packer
-->      darwin/386: github.com/hashicorp/packer
-->   windows/amd64: github.com/hashicorp/packer
-->    linux/mips64: github.com/hashicorp/packer
-->     freebsd/386: github.com/hashicorp/packer
-->    darwin/amd64: github.com/hashicorp/packer
-->     linux/s390x: github.com/hashicorp/packer
-->     openbsd/386: github.com/hashicorp/packer
==> Copying binaries for this platform...

==> Results:
total 0

$> ls pkg
darwin_386  darwin_amd64  freebsd_386  freebsd_amd64  freebsd_arm  linux_386  linux_amd64  linux_arm  linux_arm64  linux_mips  linux_mips64  linux_mipsle  linux_ppc64le  linux_s390x  openbsd_386  openbsd_amd64  windows_386  windows_amd64
```

### CODE REVIEW

**Comment by creack** [comment]
Closes #7560

File: scripts/build.sh [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 # This script builds the application from source for multiple platforms.
 # Determine the arch/os combos we're building for
-ALL_XC_ARCH="386 amd64 arm arm64 ppc64le"
-ALL_XC_OS="linux darwin windows freebsd openbsd solaris"
```
Comment (azr) [inline scripts/build.sh]: Adding those ARCH LGTM, but I'm not sure we can remove solaris support that easily.
What do you think @SwampDragons ? I think we should merge this PR like this then I can do a PR to add a `make bin` CI step and reintroduce solaris.

Comment (SwampDragons) [reply scripts/build.sh]: As long as we re-add support before we do the next release, I'm okay with that.

Comment (azr) [reply scripts/build.sh]: Got it, merging !

**Comment by james-crowley** [comment]
@azr and @creack Thanks for the help!

**Pull request merged**